int-swap-dApp/src/js/app.js

```diff
@@ -266,22 +305,45 @@ App = {
       date = `${date[1]} ${date[3]}`
 
       // proposal owner will swap out of p_swap_out_rate
-      var p_swap_out_rate = proposal_owner_struct[3];
-
-
+      p_swap_out_rate = proposal_owner_struct[3];
 
       if(counterpartyAddress == "0x0000000000000000000000000000000000000000"){
         var card = proposedIntSwapCard(p_notional_amount, p_owner_input_rate, date, p_swap_out_rate, p_escrow);
 
         $proposed_int_swap.html(card);
       }
 
-      if(counterpartyAddress != "0x0000000000000000000000000000000000000000"){
+      //show mint swap if counterparty deposited and but the intswap proposal has not created
+      if(counterpartyAddress != "0x0000000000000000000000000000000000000000" && total_escrow_deposited_in_contract){
         var mintInswapCard = intSwapCard(p_notional_amount, p_owner_input_rate, date, p_swap_out_rate, proposerAddress, counterpartyAddress, p_deposited_escrow_amount, c_deposited_escrow_amount);
-
         $mint_int_swap.html(mintInswapCard);
       }
 
+      //calculate which address belong to varToFix and vice versa
+      if(p_swap_out_rate == "variable"){ //if proposer is swapping from variable to fixed
+        variable_to_fixed_address = proposerAddress;
+        fixed_to_variable_address = counterpartyAddress;
+      } else{
+        variable_to_fixed_address = counterpartyAddress;
+        fixed_to_variable_address = proposerAddress;
+      }
+      var calculate_payout = calculatePayoutForPayees(variable_to_fixed_address, fixed_to_variable_address);
+      $("#cal_payout").html(calculate_payout);
+
+      //display the withdaw payout button
+      if($current_account == proposerAddress) {
+        //display proposer withdraw button
+        var proposer_withdraw = proposerWithdrawButton();
+        $("#withdraw").append(proposer_withdraw);
+      } else if ($current_account == counterpartyAddress) {
+          //display counterparty withdraw button
+          var counterparty_withdraw = counterpartyWithdrawButton();
+          $("#withdraw").append(counterparty_withdraw);
+      } else {
+        var not_payee = $('<h5>').attr('class', 'card-text text-danger mt-5').text(`To withdraw, you need to be either be a Proposer or Counterparty of this IntSwap contract: ${$contract_address}`);
+        $("#withdraw").append(not_payee);
+      }
+
```

Comment:
@simonnguyen3054 On lines 334-344 the $current_account refers to the first account in ganache and the account that deployed the int-swap contract and will thus not be the proposer or the counterparty, right?